language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/LSCommandExecutorProvidersHolder.java

```diff
@@ -46,12 +46,16 @@ public static LSCommandExecutorProvidersHolder getInstance(LanguageServerContext
 
     private LSCommandExecutorProvidersHolder(LanguageServerContext serverContext) {
         serverContext.put(CMD_EXEC_PROVIDERS_HOLDER_KEY, this);
+        loadServices();
+    }
+
+    private void loadServices() {
         ServiceLoader<LSCommandExecutor> loader = ServiceLoader.load(LSCommandExecutor.class);
         for (LSCommandExecutor executor : loader) {
             if (executor == null) {
                 continue;
             }
-            executors.put(executor.getCommand(), executor);
+            LSCommandExecutorProvidersHolder.executors.put(executor.getCommand(), executor);
```

Comment:
Shall we convert the static collections to instance collections?

Reply:
Nope, these collections contain SPI loaded providers. We don't need to load provider for each LanguageServerContext which will affect the performance greatly.